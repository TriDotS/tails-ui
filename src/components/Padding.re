let component = ReasonReact.statelessComponent("Padding");

let make =
    (
      ~padding,
      ~paddingLeft,
      ~paddingRight,
      ~paddingBottom,
      ~paddingTop,
      ~children
    ) => {
  ...component,
  render: _self =>
    <div
      className={j|p-$(padding) pl-$(paddingLeft) pr-$(paddingRight) pb-$(paddingBottom) pt-$(paddingTop)|j}>
      (ReasonReact.arrayToElement(children))
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~padding=jsProps##padding,
      ~paddingLeft=jsProps##paddingLeft,
      ~paddingRight=jsProps##paddingRight,
      ~paddingBottom=jsProps##paddingBottom,
      ~paddingTop=jsProps##paddingTop,
      ~children=jsProps##children
    )
  );
