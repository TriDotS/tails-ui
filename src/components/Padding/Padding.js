import PropTypes from 'prop-types'
import React from 'react'

const Padding = ({
  padding,
  paddingLeft,
  paddingRight,
  paddingBottom,
  paddingTop,
  children
}) => (
  <div
    className={`p-${padding} pl-${paddingLeft} pr-${paddingRight} pb-${paddingBottom} pt-${paddingTop}`}
  >
    {children}
  </div>
)

Padding.propTypes = {
  padding: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  paddingLeft: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  paddingRight: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  paddingBottom: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  paddingTop: PropTypes.oneOf(['0', '1', '2', '3', '4', '5', '6', '7', '8']),
  children: PropTypes.node
}

export default Padding
