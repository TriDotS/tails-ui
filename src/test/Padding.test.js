import Padding from '../components/Padding/Padding'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Padding>Some Padding</Padding>, div)
})

test('padding snapshot', () => {
  const tree = renderer.create(<Padding>Some Padding</Padding>).toJSON()
  expect(tree).toMatchSnapshot()
})
