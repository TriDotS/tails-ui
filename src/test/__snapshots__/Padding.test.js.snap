// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`padding snapshot 1`] = `
<div
  className="p-undefined pl-undefined pr-undefined pb-undefined pt-undefined"
>
  Some Padding
</div>
`;
